modules/geolocation/geolocation.py

```diff
@@ -231,20 +232,38 @@ def get_non_collinear_points(self, coordinatesArray):
             for j in range(0, NUM_POINTS_NEEDED):
                 points[j] = coordinatesArray[(i + j) % len(coordinatesArray)]
 
-            # Check collinearity of all possible combinations
             areNotFourCollinear = True
-            for i in range(0, NUM_POINTS_NEEDED):
-                areNotFourCollinear &= not self.__are_three_points_collinear(points[i],
-                                                                             points[(i + 1) % NUM_POINTS_NEEDED],
-                                                                             points[(i + 2) % NUM_POINTS_NEEDED])
+            index0 = None
```

Comment:
Why are there new variables when indexes can be used to store the current iteration? Remove this.